docs/modules/ROOT/pages/variables-and-params.adoc

```diff
@@ -0,0 +1,384 @@
+
+```javascript
+const myVar = new Cypher.Variable();
+const query = new Cypher.With(myVar).return(myVar);
+```
+
+```cypher
+WITH var0
+RETURN var0
+```
+
+Each instance you create of a variable will be associated with a different name:
+
+```javascript
+const withVar = new Cypher.Variable();
+const returnVar = new Cypher.Variable();
+const query = new Cypher.With(withVar).return(returnVar);
+```
+
+```cypher
+WITH var0
+RETURN var1
+```
+
+=== Nodes and relationships
+
+Nodes and relationships are variables holding some extra data regarding nodes and relationships to be used in xref:patterns.adoc[Patterns].
+For example:
+
+```javascript
+const movie = new Cypher.Node({labels: ["Movie"]});
+const relationship = new Cypher.Relationship({type: "ACTED_IN"});
+
+const pattern = new Cypher.Pattern(movie).related(relationship).to();
+const match = new Cypher.Match(pattern).return(movie);
+```
+
+```Cypher
+MATCH (this0:Movie)-[this1:ACTED_IN]->(this2)
+RETURN this0
+```
+
+In this example, the node variable `movie` is created and then used in the `MATCH` and `RETURN` clauses.
+
+In the translated Cypher, you can see how the node variable is translated to the Cypher variable `this0`, while the relationship is assigned to `this1`. These names are not defined in your JavaScript code, but are automatically generated when the query is built with `.build()`. The labels and type passed to `Node` and `Relationship` variables are used in the Pattern.
+
+The following parameters can be passed to Node:
+
+* `labels`: An array containing the labels of that node.
+
+The following parameters can be passed to a Relationship:
+
+* `type`: A string with the type of the relationship
+
+
+==== Escaping labels and types
+
+Labels and types will be automatically escaped if any uncommon character is detected. For example:
+
+```javascript
+const movie = new Cypher.Node({labels: ["My Movie"]});
+const match = new Cypher.Match(movie).return(movie);
+```
+
+```Cypher
+MATCH (this0:`My Movie`)
+RETURN this0
+```
+
+Note how `My Movie` is being surrounded in quotes to prevent code injection with dynamic labels.
+
+=== Path
+Path variables can be used for path assignment in patterns, for example:
+
+```cypher
+MATCH p0 = (this1)-[this2:ACTED_IN]->(this3)
+RETURN p0
+```
+
+To assign a path variable, use the method `.assignToPath` in the relevant clause and pass a `new Path()` instance:
+
+```javascript
+const path = new Cypher.Path();
+
+const query = new Cypher.Match(pattern).assignToPath(path).return(path);
+```
+
+
+=== Named Variables
+
+By default, variables do not have a name, and a unique name/id will be generated at build time to avoid collisions. However, in some cases, you may want to generate a query with a specific name. To do this, all variable types have a counterpart _named_ class:
```

Comment:
```suggestion
By default, variables do not have a name; a unique name/id is generated at build time to avoid collisions. 
However, in some cases, you may want to generate a query with a specific name. 
To do this, all variable types must have a counterpart _named_ class:
```